### PR TITLE
C build in travis

### PR DESCRIPTION
**Pull request by dkfellows**
This adds building of the C code (neural models, etc.) to our Travis test script.

It also copies a few best practices over from SpiNNFrontEndCommon.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -29,14 +25,26 @@ before_install:
   - pip install --upgrade git+git://github.com/SpiNNakerManchester/PACMAN.git@$TRAVIS_BRANCH || pip install --upgrade git+git://github.com/SpiNNakerManchester/PACMAN.git@master
   - pip install --upgrade git+git://github.com/SpiNNakerManchester/DataSpecification.git@$TRAVIS_BRANCH || pip install --upgrade git+git://github.com/SpiNNakerManchester/DataSpecification.git@master
   - pip install --upgrade git+git://github.com/SpiNNakerManchester/spalloc.git@${TRAVIS_BRANCH} || pip install --upgrade git+git://github.com/SpiNNakerManchester/spalloc.git@master
-  - pip install --upgrade git+git://github.com/SpiNNakerManchester/SpiNNFrontEndCommon.git@$TRAVIS_BRANCH || pip install --upgrade git+git://github.com/SpiNNakerManchester/SpiNNFrontEndCommon.git@master
+  - FECVER=`git ls-remote https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon.git | grep -q refs/heads/$TRAVIS_BRANCH && echo $TRAVIS_BRANCH || echo master`
+  - pip install --upgrade git+git://github.com/SpiNNakerManchester/SpiNNFrontEndCommon.git@$FECVER
+  # C dependencies
+  - git clone --branch v3.1.1 https://github.com/SpiNNakerManchester/spinnaker_tools.git
```
Comment (rowleya) [inline .travis.yml]: This should also use either the current branch or the master branch to allow synchronous branches.

File: .travis.yml [hunk 2]
```diff
@@ -29,14 +25,26 @@ before_install:
...
+  - git clone --branch 4.0.0 https://github.com/SpiNNakerManchester/spinn_common.git
```
Comment (rowleya) [inline .travis.yml]: This should also use either the current branch or the master branch to allow synchronous branches.

File: .travis.yml (hunk 1 above)
Comment (dkfellows) [reply .travis.yml]: I'm planning to leave spinnaker_tools pinned to 3.1.1 for now; that repository follows its own versioning schedule.

File: .travis.yml (hunk 2 above)
Comment (dkfellows) [reply .travis.yml]: Yes. Done

**Pull request merged**